HeterogeneousCore/CUDAServices/interface/CUDAService.h

```diff
@@ -72,6 +72,7 @@ class CUDAService {
   template <typename T>
   typename cudaserviceimpl::make_device_unique_selector<T>::non_array
   make_device_unique(cuda::stream_t<>& stream) {
+    static_assert(std::is_trivially_constructible<T>::value, "Allocating with non-trivial constructor on the device memory is not supported");
```

Comment:
Would it make sense to fail the instantiation using `std::enable_if_t<std::is_trivially_constructible<T>::value, typename ...>` as the return type, instead of using a `static_assert` ?

Pro: the templates can be used with SFINAE.
Con: the error message may be harder to understand

Reply:
I'm probably not woken up enough, but what would be the benefit of SFINAE be with these?

Reply:
Right, no, I don't think there is any advantage... 
I guess we could start playing with concepts (later).